crates/circuit/src/dag_circuit.rs

```diff
@@ -78,6 +78,15 @@ pub enum NodeType {
     Operation(PackedInstruction),
 }
 
+impl NodeType {
+    pub fn unwrap_operation(&self) -> &PackedInstruction {
+        match self {
+            NodeType::Operation(instr) => instr,
+            _ => panic!("Node is not an operation!"),
+        }
+    }
+}
```

Comment:
I may just go ahead and borrow this in my UnitarySynthesis WIP PR now because it's so much nicer than having to unwrap nodes manually (this will be merged earlier anyways).